apps/music/js/db.js

```diff
@@ -39,9 +40,14 @@ function initDB() {
   });
 
   function metadataParserWrapper(file, onsuccess, onerror) {
-    LazyLoader.load('js/metadata_scripts.js', function() {
-      AudioMetadata.parse(file).then(onsuccess, onerror);
-    });
+    LazyLoader.load(
+      ['/js/metadata_scripts.js', '/js/metadata/album_art.js'],
+      function() {
+        AudioMetadata.parse(file).then(function(metadata) {
+          AlbumArt.process(file, metadata);
+        }).then(onsuccess, onerror);
```

Comment:
AlbumArt.process() returns a promise, but the function above does not return that promise, so onsuccess or onerror is being called before the album art is processed, I think.  Is that what you intended?